nodes/core/storage/50-file.js

```diff
@@ -130,21 +135,43 @@ module.exports = function(RED) {
                             var packet = node.data.shift()
                             node.wstream.write(packet.data, function() {
                                 node.send(packet.msg);
+                                done();
                             });
 
                         } else {
                             // Dynamic filename - write and close the stream
                             var packet = node.data.shift()
                             node.wstream.end(packet.data, function() {
                                 node.send(packet.msg);
+                                delete node.wstream;
+                                delete node.wstreamIno;
+                                done();
                             });
-                            delete node.wstream;
-                            delete node.wstreamIno;
                         }
                     }
                 }
             }
+        }
+
+        function processQ(queue) {
+            var msg = queue[0];
+            processMsg(msg, function() {
+                queue.shift();
+                if (queue.length > 0) {
+                    processQ(queue);
+                }
+            });
+        }
+
+        this.on("input", function(msg) {
+            var msgQueue = node.msgQueue;
+            if (msgQueue.push(msg) > 1) {
+                // pending write exists
+                return;
+            }
+            processQ(msgQueue);
         });
+
         this.on('close', function() {
```

Comment:
When the node is closed, it needs to either clear the queue (so not all writes happen - probably the wrong answer), or block until the queue has cleared.

Reply:
fixed to wait close processing while pending messages exist.